diff --git a/lib/ansible/playbook/play_context.py b/lib/ansible/playbook/play_context.py
index a28111c..b41a996 100644
--- a/lib/ansible/playbook/play_context.py
+++ b/lib/ansible/playbook/play_context.py
@@ -81,6 +81,7 @@ MAGIC_VARIABLE_MAPPING = dict(
    su_flags         = ('ansible_su_flags',),
    executable       = ('ansible_shell_executable',),
    module_compression = ('ansible_module_compression',),
+   host_key_check   = ('ansible_host_key_check',),
 )
 
 SU_PROMPT_LOCALIZATIONS = [
@@ -199,6 +200,7 @@ class PlayContext(Base):
     _start_at_task    = FieldAttribute(isa='string')
     _step             = FieldAttribute(isa='bool', default=False)
     _diff             = FieldAttribute(isa='bool', default=False)
+    _host_key_check   = FieldAttribute(isa='bool', default=True)
 
     def __init__(self, play=None, options=None, passwords=None, connection_lockfd=None):
 
diff --git a/lib/ansible/plugins/connection/ssh.py b/lib/ansible/plugins/connection/ssh.py
index def3f85..0c6ab5b 100644
--- a/lib/ansible/plugins/connection/ssh.py
+++ b/lib/ansible/plugins/connection/ssh.py
@@ -163,6 +163,17 @@ class Connection(ConnectionBase):
                 "ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled",
                 ("-o", "StrictHostKeyChecking=no")
             )
+        else:
+            per_host_key_check = self._play_context.host_key_check
+            if not per_host_key_check:
+                self._add_args(
+                    "ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled",
+                    ("-o", "StrictHostKeyChecking=no")
+                )
+                self._add_args(
+                    "ANSIBLE_HOST_KEY_FILE/using /dev/null",
+                    ("-o", "UserKnownHostFile=/dev/null")
+                )
 
         if self._play_context.port is not None:
             self._add_args(
